src/main/java/gregtech/common/blocks/GT_Block_Machines.java

```diff
@@ -159,7 +159,33 @@ public boolean canCreatureSpawn(EnumCreatureType type, IBlockAccess aWorld, int
 
     @Override
     public boolean canConnectRedstone(IBlockAccess aWorld, int aX, int aY, int aZ, int ordinalSide) {
-        return true;
+        int forgeSide = -100;
+        switch (ordinalSide) {
+            case (-2):
+                forgeSide = 0;
+                break;
+            case (-1):
+                forgeSide = 1;
+                break;
+            case (0):
+                forgeSide = 2;
+                break;
+            case (2):
+                forgeSide = 3;
+                break;
+            case (3):
+                forgeSide = 4;
+                break;
+            case (1):
+                forgeSide = 5;
+                break;
+            default:
+                break;
+        }
+        final TileEntity machineEntity = aWorld.getTileEntity(aX, aY, aZ);
+        return machineEntity instanceof CoverableTileEntity cte
+            && cte.getCoverInfoAtSide(ForgeDirection.getOrientation(forgeSide))
+                .getCoverID() != 0;
```

Comment:
Please, use ForgeDirection directly instead of magic numbers (to improve readability).
```suggestion
        ForgeDirection forgeSide = ForgeDirection.UNKNOWN;
        switch (ordinalSide) {
            case (-2):
                forgeSide = ForgeDirection.DOWN;
                break;
            case (-1):
                forgeSide = ForgeDirection.UP;
                break;
            case (0):
                forgeSide = ForgeDirection.NORTH;
                break;
            case (2):
                forgeSide = ForgeDirection.SOUTH;
                break;
            case (3):
                forgeSide = ForgeDirection.WEST;
                break;
            case (1):
                forgeSide = ForgeDirection.EAST;
                break;
            default:
                break;
        }
        final TileEntity machineEntity = aWorld.getTileEntity(aX, aY, aZ);
        return machineEntity instanceof CoverableTileEntity cte
            && cte.getCoverInfoAtSide(forgeSide).getCoverID() != 0;
```

Reply:
Change applied.